src/prover.rs

```diff
@@ -0,0 +1,262 @@
+            .await
+        {
+            Ok(resp) => {
+                let task_input: SnarkifyCreateTaskInput = match serde_json::from_str(&resp.input) {
+                    Ok(input) => input,
+                    Err(e) => {
+                        return self.build_query_task_error_response(
+                            &req,
+                            &format!("Failed to parse task input: {}", e),
+                        )
+                    }
+                };
+                let started_at = resp.started.map(|t| t.timestamp() as f64);
+                let finished_at = resp.finished.map(|t| t.timestamp() as f64);
+                let compute_time_sec = match (started_at, finished_at) {
+                    (Some(started), Some(finished)) => Some(finished - started),
+                    _ => None,
+                };
+                QueryTaskResponse {
+                    task_id: resp.task_id,
+                    circuit_type: task_input.circuit_type,
+                    circuit_version: task_input.circuit_version,
+                    hard_fork_name: task_input.hard_fork_name,
+                    status: resp.state.into(),
+                    created_at: resp.created.map(|t| t.timestamp() as f64).unwrap_or(0.0),
+                    started_at,
+                    finished_at,
+                    compute_time_sec,
+                    input: Some(task_input.task_data),
+                    proof: resp.proof,
+                    vk: None,
+                    error: resp.error,
+                }
+            }
+            Err(e) => {
+                error!("query_task method failed: {:?}", e);
+                self.build_query_task_error_response(&req, &format!("Failed to query proof: {}", e))
+            }
+        }
+    }
+}
+
+impl SnarkifyProver {
+    pub fn new(cfg: CloudProverConfig, service_id: String) -> Self {
+        let retry_wait_duration = Duration::from_secs(cfg.retry_wait_time_sec);
+        let retry_policy = ExponentialBackoff::builder()
+            .retry_bounds(retry_wait_duration / 2, retry_wait_duration)
+            .build_with_max_retries(cfg.retry_count);
+        let client = ClientBuilder::new(reqwest::Client::new())
+            .with(RetryTransientMiddleware::new_with_policy(retry_policy))
+            .build();
+
+        Self {
+            base_url: cfg.base_url,
+            api_key: cfg.api_key,
+            service_id,
+            send_timeout: Duration::from_secs(cfg.connection_timeout_sec),
+            client,
+        }
+    }
+
+    pub fn build_prove_error_response(&self, req: &ProveRequest, error_msg: &str) -> ProveResponse {
+        ProveResponse {
+            task_id: String::new(),
+            circuit_type: req.circuit_type,
+            circuit_version: req.circuit_version.clone(),
+            hard_fork_name: req.hard_fork_name.clone(),
+            status: TaskStatus::Failed,
+            created_at: 0.0,
```

Comment:
why is it 0.0 for a datetime?

Reply:
it is float type in `ProveResponse`, which is defined in scroll proving sdk